src/main/java/rcms/utilities/daqaggregator/data/RU.java

```diff
@@ -307,6 +309,34 @@ public void setIncompleteSuperFragmentCount(int incompleteSuperFragmentCount) {
 		this.incompleteSuperFragmentCount = incompleteSuperFragmentCount;
 	}
 
+	/** @return the list of all (real) FEDs (not including pseudofeds) associated
+	 *  to this RU.
+	 *  @param includeMasked if true, includes also masked FEDs, otherwise
+	 *    just includes FEDs which are not masked FRL-wise.
+	 */
+	public Set<FED> getFEDs(boolean includeMasked) {
+
+		Set<FED> retval = new HashSet<>();
+
+		for (SubFEDBuilder subFedBuilder : this.getFedBuilder().getSubFedbuilders()) {
```

Comment:
The same may happen here - Fed has no fed builder for some reason, or Fed builder no subfed builder. I suggest we put the exception handing here as well.